File: vuln-sandbox/ruby/cwe-79/xss-safe.rb
# CWE-79: Cross-Site Scripting - Safe Pattern
# Rule: ruby-xss-html-safe
# Expected: TrueNegative

class CommentsController < ApplicationController
  def show
    # SAFE: Rails auto-escapes output in ERB templates
    @comment = params[:comment]
    render inline: "<p><%= @comment %></p>"
  end

  def preview
    # SAFE: Using sanitize helper to allow only safe HTML
    content = sanitize(params[:content], tags: %w[p b i em strong])
    render html: content.html_safe
  end
end

File: vuln-sandbox/php/cwe-79/xss-safe.php
<?php
// CWE-79: Cross-Site Scripting - Safe Pattern
// Rule: php-xss-echo-user-input
// Expected: TrueNegative

// SAFE: HTML-encode user input before output
echo htmlspecialchars($_GET['name'], ENT_QUOTES, 'UTF-8');

// SAFE: Using htmlentities for full encoding
echo htmlentities($_POST['message'], ENT_QUOTES, 'UTF-8');

// SAFE: Static content, no user input
echo "Welcome to our application!";
?>
